.rubocop.yml

```diff
@@ -25,6 +25,35 @@ Style/Documentation:
 Style/ClassAndModuleChildren:
   Enabled: false
 
+# This cop forces you to put a return at the beginning of a block of code rather than having an if statement
+# whose body carries to the end of the function. For example:
+#
+# def foo
+#   ...
+#   if test
+#     ...
+#   end
+# end
+#
+# would be considered bad, and the cop would force you to put a `return if !test` before that block and
+# then remove the if. The problem is that this hides intent, since the if test does have a purpose in
+# readability, and it could also be easier for future changes to miss the return statement and add code
+# after it expecting it to be executed.
+Style/GuardClause:
+  Enabled: false
+
+# This is pretty much the same thing as the one above. Inside a loop, it forces you to use next to skip
+# iteration rather than using an if block that runs to the end of the loop, and it suffers from the same
+# problems as above.
+Style/Next:
+  Enabled: false
```

Comment:
@saneshark The one with `next` is the same as the guard clause one except for within a loop. So, if you wrote code like this:

```ruby
foos.each do |foo|
  puts foo + 2
  if foo > 0
    command1(foo)
    command2(foo)
    command3(foo)
  end
end
```

It would require you to change it to:

```ruby
foos.each do |foo|
  puts foo + 2
  next if foo <= 0
  command1(foo)
  command2(foo)
  command3(foo)
end
```

And I think it suffers from the same problems. Namely, the `next` hides the intent that is exposed with the `if` and it makes maintenance more difficult.